freedomofpress/securedrop
# Remove ntp and ntpdate dependencies

## Status

Ready for review

## Description of Changes

On Focal, stop installing the `ntp` and `ntpdate` packages, and update the system time with `systemd-timesyncd`.

Fixes #5730.
Fixes #5795.

I also updated `molecule/testinfra/conftest` to make the class that provides attribute access to test variables aware of distribution-specific variables, such that you can ask for `apparmor_enforce`, and if that's a dictionary containing your target distribution, e.g. `focal`, it will return that item from the dictionary.

Finally, there were some complete iptables rulesets in the testinfra vars that didn't seem to be used anywhere, so I removed those.

## Testing

- `make build-debs-focal && make staging-focal`
- confirm that the `ntp` and `ntpdate` packages are not installed on either server
- confirm that time has been synchronized to NTP servers on both machines:
  - `timedatectl show` should contain `NTPSynchronized=yes`
  - `timedatectl show-timesync` should contain `ServerName=ntp.ubuntu.com`, with an `NTPMessage` indicating that the server has been reached

## Deployment

For Focal systems, this replaces the `universe` packages `ntp` and `ntpdate` with the `admin` package `systemd-timesyncd`.

The iptables rules for NTP required changes:
- `systemd-timesyncd` does not control its source port
- it also runs as the `systemd-timesync user`

## Checklist

### If you made changes to the server application code:


### If you made changes to the system configuration:


### If you made non-trivial code changes:


Choose one of the following:
